### PR TITLE
new pipeline framework

### PR DESCRIPTION
**Pull request by bheni**

### CODE REVIEW

**Comment by bheni** [comment]
```
+------------+------+------+--------------+		
|            |  old |  new |  improvement |
+------------+------+------+--------------+
| Tabular    | 1773 | 1338 |     24.5%    |
| csv        |  671 |  480 |     28.4%    |
| json       | 1004 |  575 |     42.7%    |
+------------+------+------+--------------+
```

File: go/libraries/utils/pipeline/local_storage.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+// Put stores an item in local storage
+func (ls LocalStorage) Put(id int, val interface{}) {
+	ls[id] = val
+}
+
+// GetLocalStorage retrieves the LocalStorage from the context.  This only works if the context was generated
+// by the pipeline package when starting the pipeline
+func GetLocalStorage(ctx context.Context) LocalStorage {
```
Comment (zachmu) [inline go/libraries/utils/pipeline/local_storage.go:37]: This whole thing is kind of gross. It's only used one place outside tests. Can we get rid of it?

**Review by zachmu — APPROVED** [review]
Generally looks quite good, just a few comments.

File: go/libraries/utils/pipeline/local_storage.go (hunk 1 above)
Comment (bheni) [reply go/libraries/utils/pipeline/local_storage.go:37]: Yeah.  it was mostly something I was playing with as a possible way to optimize in the future.  I don't have a case where it really makes a difference atm.

Comment (bheni) [reply go/libraries/utils/pipeline/local_storage.go:37]: added a todo.  will revisit post TG.

**Pull request merged**